### DISCUSSION TITLE
error: exportArchive: No Team Found in Archive

### DISCUSSION THREAD

**Discussion by developer1-miksi** [codemagic-ci-cd/codemagic-docs#2482] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

the build succeeds and it shows that iOS signing is correct 
Generated options for exporting the project
 - Method: app-store
 - Provisioning Profiles:
     - com.mycompany.client1
 - Signing Certificate: Apple Distribution (uploaded certificate to code magic)
 - Signing Style: manual
 - Team Id: (it shows the correct team id too )
 - and here's when I try to build IPA for flutter and push it to App Store 
 `Archiving com.mycompany.client1...
Upgrading contents.xcworkspacedata
Updating project for Xcode compatibility.
Upgrading project.pbxproj
Upgrading Runner.xcscheme
Upgrading Info.plist
Removing script build phase dependency analysis.
Adding input path to Thin Binary build phase.
Automatically signing iOS for device deployment using specified development team in Xcode project: G344VHW6HF
Running pod install...                                           1,573ms
Running Xcode build...                                          
Xcode archive done.                                         68.5s
Built /Users/builder/clone/build/ios/archive/Runner.xcarchive.

[✓] App Settings Validation
    • Version Number: 1.0.11
    • Build Number: 11
    • Display Name: TestAppClient
    • Deployment Target: 13.0
    • Bundle Identifier: com.mycompany.client1

[!] App Icon and Launch Image Assets Validation
    ! Launch image is set to the default placeholder icon. Replace with unique launch image.

To update the settings, please refer to https://docs.flutter.dev/deployment/ios

Building App Store IPA...                                          511ms
Encountered error while creating the IPA:
error: exportArchive: No Team Found in Archive

Try distributing the app in Xcode: "open /Users/builder/clone/build/ios/archive/Runner.xcarchive"`

here's my Codemagic.yaml file 
`definitions:
  env_versions: &env_versions
    flutter: stable
    xcode: 14.2
    cocoapods: default

    # FIREBASE_PROJECT_ID: your-firebase-project-id
    # BASE_ASSETS_URL: your-base-assets-url # This might not be needed if ASSETS_URL is fully specified per build
    # ASSETS_URL: "https://example.com/path/to/assets_${CLIENT_ID}.zip" # This should be set dynamically per build

workflows:
  ios-client-release:
    name: iOS client release
    instance_type: mac_mini_m1
    labels:
      - ${CLIENT_ID}
    environment:
      groups:
        - env_vars # Ensure this group includes CLIENT_ID and ASSETS_URL
      ios_signing:
        # team_id: "G344VHW6HF"
        provisioning_profiles:
          - profile: clientApp1
        certificates:
          - certificate: distributionCertificate
      vars:
        XCODE_SCHEME: "Runner"
        # FIREBASE_TOKEN: ${FIREBASE_TOKEN}
        # APP_NAME: ${APP_NAME}
        # BUNDLE_ID: ${BUNDLE_ID}
    scripts:

     - name: Debug environment variables
        script: |
          echo "App Name: ${APP_NAME}"
          echo "Bundle ID: ${BUNDLE_ID}"
          echo "Firebase Token: ${FIREBASE_TOKEN}"

      - name: Download and unpack client-specific assets
        script: |
          echo "Downloading assets from ${ASSETS_URL}"
          curl -o "assets_${CLIENT_ID}.zip" "${ASSETS_URL}" || echo "Failed to download assets_${CLIENT_ID}.zip"
          echo "Removing old assets and configuration files"
          rm -rf ./assets/
          unzip "assets_${CLIENT_ID}.zip" -d ./assets
          echo "Adjusting assets directory structure"
          mv ./assets/assets/* ./assets/
          rm -rf ./assets/assets
          echo "List of files in assets directory:"
          ls -la ./assets/

      - name: Update assets and configuration files
        script: |
          echo "Replacing Google services files"
          if [ -f "./assets/google-services.json" ]; then
            cp ./assets/google-services.json ./android/app/
          else
            echo "google-services.json not found in assets folder."
          fi
          if [ -f "./assets/GoogleService-Info.plist" ]; then
            cp ./assets/GoogleService-Info.plist ./ios/Runner/
          else
            echo "GoogleService-Info.plist not found in assets folder."
          fi
          if [ -f "./assets/client/client.env" ]; then
            cp ./assets/client/client.env ./assets/
          else
            echo "client.env file not found in assets/client folder."
          fi

          flutter pub get
          flutter pub run flutter_launcher_icons

      - name: Change iOS app name
        script: |
          /usr/libexec/PlistBuddy -c "Set :CFBundleName $APP_NAME" -c "Set :CFBundleDisplayName $APP_NAME" ios/${XCODE_SCHEME}/Info.plist

      - name: Set bundle id
        script: |
          sed -i '' -e "s/PRODUCT_BUNDLE_IDENTIFIER \= [^\;]*\;/PRODUCT_BUNDLE_IDENTIFIER = '${BUNDLE_ID}';/" ios/${XCODE_SCHEME}.xcodeproj/project.pbxproj
          echo "Bundle ID set to: ${BUNDLE_ID}"

      - name: Install dependencies
        script: |
          flutter packages pub get

      # - name: Set up code signing settings on Xcode project
      #   script: |
      #     xcode-project use-profiles
      - name: iOS code signing
        script: |
          keychain initialize
          app-store-connect fetch-signing-files "$BUNDLE_ID" --type IOS_APP_STORE --create
          keychain add-certificates
          xcode-project use-profiles

      - name: Install dependencies
        script: flutter packages pub get

      - name: Flutter build ipa and automatic versioning
        script: |
          BUILD_NUMBER=$(($(app-store-connect get-latest-testflight-build-number "$APP_STORE_ID") + 1))
          flutter build ipa --release \
          --build-name=1.0.$BUILD_NUMBER \
          --build-number=$BUILD_NUMBER\
          --export-options-plist=/Users/builder/export_options.plist
    artifacts:
      - build/ios/ipa/*.ipa
      - /tmp/xcodebuild_logs/*.log
      - test-results/flutter.json
      - $HOME/Library/Developer/Xcode/DerivedData/**/Build/**/*.dSYM
    publishing:
      app_store_connect:
        api_key: $APP_STORE_CONNECT_PRIVATE_KEY
        key_id: $APP_STORE_CONNECT_KEY_IDENTIFIER
        issuer_id: $APP_STORE_CONNECT_ISSUER_ID
      email:
        recipients:
          - name@mail.com
`

### Expected results

it should push the archived and signed IPA to App Store 

### Actual results

Building App Store IPA...
Encountered error while creating the IPA:
error: exportArchive: No Team Found in Archive

### Build id (optional)

661f3e812a75c9d00fab1ade

**Comment by dtrdic** [comment]
Note: conversation continued on the Intercom chat

**Reply by dtrdic** [reply, marked as answer]
Note: issue resolved
```
     BUILD_NUMBER=$(($(app-store-connect get-latest-testflight-build-number "$APP_STORE_ID") + 1))
      flutter build ipa --release \
      --build-name=1.0.$BUILD_NUMBER \
      --build-number=$BUILD_NUMBER\ <-- this backslash was causing a syntax error, it has to have empty space before \
      --export-options-plist=/Users/builder/export_options.plist
```

**Reply by billdacat** [reply]
I'm so glad I stumbled upon this -- this is still an issue in the documentation/example code and was tripping me up for some time.  Thanks for posting the solution here and hopefully the docs get fixed.
